### PR TITLE
Enable configuration caching

### PR DESCRIPTION
**Pull request by clayburn**

### CODE REVIEW

**Review by runningcode — APPROVED** [review]
Nice!

File: build.gradle [hunk 1]
```diff
@@ -97,6 +97,15 @@ tasks.register('createReleaseTag', CreateGitTag) {
...
 tasks.named('githubRelease') {
     dependsOn('createReleaseTag')
+    notCompatibleWithConfigurationCache("$name task does not support configuration caching")
+}
+
+tasks.named('publishPlugins') {
+    notCompatibleWithConfigurationCache("$name task does not support configuration caching")
+}
+
+tasks.withType(Sign) {
```
Comment (clayburn) [inline build.gradle:107]: Made this more generic to account for upcoming changes in the `plugin-publish` plugin.

File: build.gradle [hunk 2]
```diff
@@ -97,6 +97,15 @@ tasks.register('createReleaseTag', CreateGitTag) {
 
 tasks.named('githubRelease') {
     dependsOn('createReleaseTag')
+    notCompatibleWithConfigurationCache("$name task does not support configuration caching")
```
Comment (etiennestuder) [inline build.gradle:100]: I believe this line can be removed now that the new GH plugin release is out and applied to this build. Please verify by publishing a new version to the dev portal.

File: build.gradle [hunk 3]
```diff
@@ -97,6 +97,15 @@ tasks.register('createReleaseTag', CreateGitTag) {
...
+}
+
+tasks.named('publishPlugins') {
+    notCompatibleWithConfigurationCache("$name task does not support configuration caching")
+}
```
Comment (etiennestuder) [inline build.gradle:105]: Will it be CC compatible in version 1.0 of the publish plugin?

File: build.gradle [hunk 4]
```diff
@@ -97,6 +97,15 @@ tasks.register('createReleaseTag', CreateGitTag) {
...
+
+tasks.withType(Sign).configureEach {
+    notCompatibleWithConfigurationCache("$name task does not support configuration caching")
```
Comment (etiennestuder) [inline build.gradle:108]: Do you know the plans for making it CC comaptible?

**Review by etiennestuder — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 3 above)
Comment (clayburn) [reply build.gradle:104]: It does not appear to be cacheable on the current release candidate.

File: build.gradle (hunk 4 above)
Comment (clayburn) [reply build.gradle:107]: I don't, but it appears to be tracked by: [Make `signing` plugin support the configuration cache · Issue #13470 · gradle/gradle](https://github.com/gradle/gradle/issues/13470)

**Review by etiennestuder — APPROVED** [review]

**Pull request merged**